### DISCUSSION TITLE
Adding extra functionality to analysis/replay mode

### DISCUSSION THREAD

**Discussion by aMalteser** [ppy/osu#29767] [Ideas]
The addition of the analaysis tools has been a great feature; but, with addition with these 3 extra pieces of functionality the experience using these tools can be greatly improved.

## Ability to quickly exit and review current play

- Add a keybinding that works similarly to quick retry that allows the user to quit out the play and instantly view the replay

Currently the only way to be able to analyise your play is to complete the play or to purposefully fail. I think adding a keybinding that allows you to instantly view the replay

## Add judgement markers to argon timeline

- In the argon timeline add red bars to show where the user has broken combo (could also include other coloured bars for 100s, 50s, missed slider ticks, etc.)

![image](https://github.com/user-attachments/assets/2aa36853-851a-4d51-881d-e817aec3aab6)
(quick sketchup)

Displaying to the user where misses (or possibly other misjudgements) occured will quickly identify the points of interest for analyising the replay in the first place

## Reworking the handling of failed replays

- Don't exit replays after failure

Self explanatory, it is a frustration trying to seek through a failed replay to get to the section that you want to view, only to go too far and be forced out of the replay

**Comment by peppy** [comment]
All these suggestions sound good. The last one is especially important IMO.

**Reply by Tenexxt** [reply]
I'm don't know what's going to be the approach on failed replays, but personally it would make sense to me for the replay to cut a few seconds after you died to save data, because why would you save a 5min of basically nothing on a 6min map after you died on the first one.
The few second delay after death would be there so you can still analyse those final notes with higher precision.
